cs/testing-with-travis.texy

```diff
@@ -0,0 +1,192 @@
+
+Bloky jsou spouštěny v pořadí `before_install`, `install`, `before_script`, `script`, `after_success` nebo `after_failure` a nakonec `after_script`. Povinný je pouze blok `script`.
+
+K výsledku testu můžete v nastavení přistoupit pomocí proměnné `$TRAVIS_TEST_RESULT`.
+
+Pro podrobnější informace si přečtěte [oficiální dokumentaci |http://docs.travis-ci.com].
+
+Nastavení jazyka
+================
+
+Nejprve je potřeba pomocí `language` Travis informovat, pro jaký jazyk má prostředí nastavit. Na dalších řádcích uvádíme, pro jaké verze PHP budeme chtít náš projekt otestovat. Testy pak budou puštěny pro každou verzi zvlášť. Protože Nette Tester podporuje i [HHVM](http://hhvm.com), můžeme ho přidat také.
+
+/--
+language: php
+
+php:
+  - 5.3
+  - 5.4
+  - 5.5
+  - 5.6
+  - hhvm
+\--
+
+Proměnné prostředí
+==================
+
+V sekci `env` nastavujeme proměnné prostředí. Pro každou odrážku se skripty provedou samostatně a v jedné odrážce lze definovat více proměnných. My budeme používat proměnnou `TESTER_PHP_BIN`, jejíž hodnotu poté získáme pomocí prefixu `$`. 
+
+
+
+/--
+env:
+  - TESTER_PHP_BIN=php
+  - TESTER_PHP_BIN=php-cgi
+\--
+
+Instalace závislostí
+====================
+
+.[caution]
+Návod předpokládá, že závislosti vašeho projektu jsou spravovány pomocí nástroje Composer a v kořenovém adresáři obsahuje validní `composer.json`. Zároveň musíte mít v závislostech uvedený nette/tester. Jak nastavit Composer najdete v kapitole [#Composer].
+
+V bloku `install` nainstalujeme pomocí Composeru závislosti našeho projektu. Každá odrážka znamená samostatný příkaz. Poté nainstalujeme závilosti včetně těch vývojářských `--dev`. Protože nechceme, aby se závislosti načítali z keše, použijeme `--prefer-source`. A nahonec, protože nechceme aby composer pokládal žádné interaktivní dotazy, na které by Travis nedokázal odpověděť, použijeme `--no-interaction`.
+
+
+Protože chceme používat aktuální verzi composeru, provedeme v bloku `before_install` jeho update.
+
+/--
+before_install:
+  - composer self-update
+
+install:
+  - composer install --no-interaction --prefer-source --dev
+\--
+
+Matice testů
+============
+
+Pro každý projekt je na základě předchozích údajů vytvořena testovací matice, tedy soubor testů, které budou spuštěny a obsahují veškeré možná kombinace. Pokud chceme některé kombinace vyřadit, můžeme využít blok `matrix`.
+Například `php-cgi` nebude fungovat pro HHVM, ani pro verzi 5.3.
+
+
+/--
+matrix:
+  exclude:
+    - php: hhvm
+      env: TESTER_PHP_BIN="php-cgi"
+    - php: 5.3
+      env: TESTER_PHP_BIN="php-cgi"
+\--
+
+Pokud některé testy chceme pustit, ale nevadí nám, že neprojdou, můžeme to Travisu sdělit pomocí `allow_failures`:
+
+/--
+matrix:
+  allow_failures:
+    - php: hhvm
+\--
+
+Travis nyní nebude hlásit selhání, pokud selže hhvm, ale ostatní testy procházejí.
```

Comment:
Dost divná formulace. Mělo by to znít spíš: "... pokud některé testy selžou pod HHVM, ale pod jinými ...".